src/main/java/com/example/bookhub/service/UserServiceImpl.java

```diff
@@ -22,7 +29,14 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto)
             throw new RegistrationException("User with this email: "
                     + requestDto.getEmail() + " already exist");
         }
-        User savedUser = userRepository.save(userMapper.toModel(requestDto));
+        User userToSave = userMapper.toModel(requestDto);
+        userToSave.setPassword(passwordEncoder.encode(requestDto.getPassword()));
+
+        Role userRole = roleRepository.findByName(RoleName.USER)
+                .orElseThrow(() -> new RuntimeException("Role USER not found"));

```

Comment:
add RoleName.USER to message instead of USER
```suggestion
                .orElseThrow(() -> new EntityNotFoundException("Role USER not found"));
```